lib/ublksrv.c

```diff
@@ -937,7 +937,7 @@ int ublksrv_process_io(const struct ublksrv_queue *tq)
 				q->cmd_inflight, q->tgt_io_inflight,
 				(q->state & UBLKSRV_QUEUE_STOPPING));
 
-	if (ublksrv_queue_is_done(q))
+	if (ublksrv_queue_is_done(tq))
```

Comment:
ublksrv_queue_is_done() is run in fast path, and it is inlined always.  I'd suggest to add one inline function __ublksrv_queue_is_done() for ublksrv_process_io(), and meantime call this inline from the exported ublksrv_queue_is_done().

Reply:
done